src/lamp_py/tableau/jobs/bus_performance.py

```diff
@@ -76,7 +78,29 @@ def create_bus_parquet(job: HyperJob, num_files: Optional[int]) -> None:
 
     with pq.ParquetWriter(job.local_parquet_path, schema=job.parquet_schema) as writer:
         for batch in ds.to_batches(batch_size=500_000):
-            writer.write_batch(batch)
+            arrow_table = pyarrow.Table.from_batches([batch])
+
+            polars_df = pl.from_arrow(arrow_table)
+            # convert to df if series
+            if isinstance(polars_df, pl.Series):
+                polars_df = polars_df.to_frame()
```

Comment:
I know this is a type guard, forced by mypy, but if `polars_df` does happen to be a `Series` instead of a `DataFrame`, something has gone horribly wrong, this if check should raise some kind of Exception instead of converting to a DataFrame.